systemd/datadog_checks/systemd/systemd.py

```diff
@@ -0,0 +1,175 @@
+
+        for unit in list_unit_files:
+            unit_short_name = unit[0]  # unit name/id
+            unit_state = unit[3]  # unit state
+            unit_status[unit_short_name] = unit_state
+
+        return unit_status
+
+    def list_status_change(self, current_unit_status):
+        # returns a tuple of dicts for unit changes
+        changed = defaultdict(dict)
+        deleted = defaultdict(dict)
+        created = copy.deepcopy(current_unit_status)
+
+        for unit_short_name, previous_unit_status in iteritems(self.unit_cache):
+            # We remove all previous cached entries from the current unit status
+            created.pop(unit_short_name, None)
+            # We check status changes between the previous cached unit status and the current unit status
+            current_status = current_unit_status.get(unit_short_name)
+            if current_status:
+                if current_status != previous_unit_status:
+                    self.log.debug("unit {} changed state, it is now {}".format(unit_short_name, current_status))
+                    changed[unit_short_name] = current_status
+            else:
+                self.log.debug("unit {} not found".format(unit_short_name))
+                # We list all previous cached unit status that do not exist anymore
+                deleted[unit_short_name] = previous_unit_status
+
+        return changed, created, deleted
+
+    def report_changed_units(self, units, tags):
+        for unit, state in iteritems(units):
+            self.event({
+                "event_type": "unit.status.changed",
+                "msg_title": "unit {} changed state".format(unit),
+                "msg_text": "it is now: {}".format(state),
+                "tags": tags
+            })
+
+    def report_deleted_units(self, units, tags):
+        for unit, state in iteritems(units):
+            self.event({
+                "event_type": "unit.status.deleted",
+                "msg_title": "unit {} cannot be found".format(unit),  # TODO: check wording
+                "msg_text": "last reported status was: {}".format(state),
+                "tags": tags
+            })
+
+    def report_created_units(self, units, tags):
+        for unit, state in iteritems(units):
+            self.event({
+                "event_type": "unit.status.created",
+                "msg_title": "new unit {} has been found".format(unit),  # TODO: check wording
+                "msg_text": "status is: {}".format(state),
+                "tags": tags
+            })
+
+    def report_statuses(self, units, tags):
+        active_units = inactive_units = 0
+        for unit, state in iteritems(units):
+            state = ensure_unicode(state)
+            try:
+                if state == 'active':
+                    active_units += 1
+                if state == 'inactive':
+                    inactive_units += 1
+            except pystemd.dbusexc.DBusInvalidArgsError as e:
```

Comment:
Is this ever triggered? The lines in the try block are checking string equality? Are there any systemd calls being made?